### PR TITLE
Split up main page into sections.

### PR DESCRIPTION
**Pull request by wvreeven**

### CODE REVIEW

File: doc/development/building.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Building Applications
+=====================
+
+`Jenkins <https://tssw-ci.lsst.org/>`_ is the chosen Continuous Integration platform.
+Each application should have a build project in Jenkins.
```
Comment (tribeiro) [inline doc/development/building.rst:5]: I think you could turn this into a bullet point

File: doc/development/building.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+These applications should then have a dedicated build server.
```
Comment (tribeiro) [inline doc/development/building.rst:6]: No sure I understand what this phrase means... Are you trying to say we have to select one of the Jenkins nodes to run the CI? I think in general we leave it float to the nodes so Jenkins can schedule them at will, except for the conda recipes, which have a dedicated node.

File: doc/development/building.rst [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+Each build should run the unit tests.
+If the unit tests pass, the application should generate the deployable package and feed into the deployment system.
+Puppet is the Deployment application.
```
Comment (tribeiro) [inline doc/development/building.rst:10]: This is not strictly correct.. I think only a handful of applications use Puppet. In general we use Puppet to provision and manage the infrastructure and k8s to deploy. 

In any case, this is talking about "Build" so maybe remove this part about deployment.

File: doc/development/building.rst [hunk 4]
```diff
@@ -0,0 +1,16 @@
+Building Applications
+=====================
+
+`Jenkins <https://tssw-ci.lsst.org/>`_ is the chosen Continuous Integration platform.
+Each application should have a build project in Jenkins.
+These applications should then have a dedicated build server.
+
+Each build should run the unit tests.
+If the unit tests pass, the application should generate the deployable package and feed into the deployment system.
```
Comment (tribeiro) [inline doc/development/building.rst:9]: Consider rewording this:

> If the build is successful (including the unit tests), it should generate artifacts that can either be directly deployed  or used by other upstream builds to generate deployable artifacts.

File: doc/development/building.rst [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+Puppet is the Deployment application.
+See `Puppet Server Installation <https://confluence.lsstcorp.org/display/~avillalo/Puppet+Server+Installation>`_ and https://puppet.com/ for more information.
+
+The packaging system is still being investigated, but is leaning towards using RPMs, in general.
```
Comment (tribeiro) [inline doc/development/building.rst:13]: I think you can reword this paragraph to reflect the current reality:

> Artifacts should be packaged with one of the following adopted solutions:
> * Python software should be packed with conda.
> * RPM is a acceptable for C++ and other types of applications (e.g. the middleware).
> * Java code should use maven. (not sure this is true, maybe check with Dave what he does for Tony).
> * Labview is still TBD.
> * For exceptions an RFC ticket is required.

File: doc/procedures/jira_workflow.rst [hunk 6]
```diff
@@ -0,0 +1,86 @@
+Jira workflow
```
Comment (tribeiro) [inline doc/procedures/jira_workflow.rst:1]: Maybe rename this "Development Workflow"?

File: doc/procedures/jira_workflow.rst [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+* Write the code.
+* Write the unit tests.
+* Update documentation
+* When complete, move to In Review.
+
+  * Complete meaning:
+
+    * Add a link of the PR to the ticket.
+    * Unit tests exist, have been successfully run and results have been added to the ticket or PR.
+    * Add a link to the generated documentation site
```
Comment (tribeiro) [inline doc/procedures/jira_workflow.rst:48]: missing a dot at the end of this line

File: doc/procedures/jira_workflow.rst [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  * Code is complete and understandable.
+  * CI is passing.
+  * Documentation is done, including a reference to the lsst.io site
+  * All requirements, as defined in the ticket description, are met.
+
+* If issues are found
+
+  * Update Jira with issue
+
+    * Comments on GitHub PR are considered acceptable as well
```
Comment (tribeiro) [inline doc/procedures/jira_workflow.rst:65]: I think comments in the PR is preferable no?

File: doc/procedures/jira_workflow.rst [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+  * Sends back to developer
+
```
Comment (tribeiro) [inline doc/procedures/jira_workflow.rst:68]: We may want to add a phrase or two about conflict resolution, when the developer and the reviewer don't see eye to eye... Maybe add:

> * If conflicts arise during the review process, developers should reach out to the Scrum Master and/or the Software Architect to seek a conflict resolution.
>  If a resolution can't be found the issue can be escalated further to the Product Owner and the  Software Manager.

File: doc/procedures/jira_workflow.rst [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+* If no issues are found
+
+  * Moves Jira ticket to review complete with approval/minor changes.
```
Comment (tribeiro) [inline doc/procedures/jira_workflow.rst:71]: > * The Reviewer approves the PR and moves the Jira ticket to Reviewed with approval/minor changes requested.

File: doc/procedures/jira_workflow.rst [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+Reviewed
+--------
+
+* The Developer then merges the pull-request.
```
Comment (tribeiro) [inline doc/procedures/jira_workflow.rst:76]: > * The Developer has a chance to apply minor changes and merges the pull-request.

File: doc/procedures/jira_workflow.rst [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+* Moves ticket to Done.
```
Comment (tribeiro) [inline doc/procedures/jira_workflow.rst:77]: > * The Developer moves the ticket to Done.

File: doc/procedures/versioning.rst [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+  * ticket branches can be tagged with ``vX.Y.Z.alpha.N`` or ``vX.Y.Z.beta.N`` tags
+  * The develop branch can be tagged with ``vX.Y.Z.rc.N``
+    If your code is using ts_xml develop branch then the CSC release must be tagged here.
+  * Main branch is reserved for main tags: ``vX.Y.Z``
+    Any releases tagged from here must be compatible with the current released version of ts_xml.
+
+Where X, Y and Z are major, minor and point/hotfix respectively.
+
+  * Proposed definitions for Major, Minor and Point/Hotfix: https://semver.org/
```
Comment (tribeiro) [inline doc/procedures/versioning.rst:14]: Move this to the top and rephrase to enforce the rule.

> Package versioning should use `Semantic Versioning <https://semver.org/>` _.
> Exceptions (e.g. see the cycle build) should be well documented in the package and may require an RFC ticket.

**Review by tribeiro — APPROVED** [review]
Looks great! Thanks for the update! I made a few inline comments I hope you will consider.. I know you were mostly reorganizing it but I think we could take the opportunity to review/update some of the information here. I left mostly direct copy/paste text you can use.

File: doc/development/building.rst (hunk 1 above)
Comment (wvreeven) [reply doc/development/building.rst:5]: Done

File: doc/development/building.rst (hunk 2 above)
Comment (wvreeven) [reply doc/development/building.rst:6]: I didn't make any changes to this file, I merely moved text around. So I don't understand the sentence either hehe. However, I modified the sentence to reflect the fact that Jenkins chooses the node to run the build on.

File: doc/development/building.rst (hunk 3 above)
Comment (wvreeven) [reply doc/development/building.rst:10]: Done

File: doc/development/building.rst (hunk 4 above)
Comment (wvreeven) [reply doc/development/building.rst:9]: Done

File: doc/development/building.rst (hunk 5 above)
Comment (wvreeven) [reply doc/development/building.rst:13]: Good suggestion, thanks. Done.

File: doc/procedures/jira_workflow.rst (hunk 6 above)
Comment (wvreeven) [reply doc/procedures/jira_workflow.rst:1]: Done

File: doc/procedures/jira_workflow.rst (hunk 7 above)
Comment (wvreeven) [reply doc/procedures/jira_workflow.rst:48]: Done

File: doc/procedures/jira_workflow.rst (hunk 8 above)
Comment (wvreeven) [reply doc/procedures/jira_workflow.rst:65]: Done

File: doc/procedures/jira_workflow.rst (hunk 9 above)
Comment (wvreeven) [reply doc/procedures/jira_workflow.rst:68]: Done

File: doc/procedures/jira_workflow.rst (hunk 10 above)
Comment (wvreeven) [reply doc/procedures/jira_workflow.rst:71]: Done

File: doc/procedures/jira_workflow.rst (hunk 11 above)
Comment (wvreeven) [reply doc/procedures/jira_workflow.rst:76]: Done

File: doc/procedures/jira_workflow.rst (hunk 12 above)
Comment (wvreeven) [reply doc/procedures/jira_workflow.rst:77]: Done

File: doc/procedures/versioning.rst (hunk 13 above)
Comment (wvreeven) [reply doc/procedures/versioning.rst:14]: Done

**Pull request merged**